bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/Utils.java

```diff
@@ -90,12 +94,15 @@ public static String obfuscateToken(String tokenString) {
         }
     }
 
-    public static JsonObject convertFileToJSON(URL fileName) throws JsonIOException, JsonSyntaxException, IOException {
+    public static JsonObject convertFileToJSON(URL fileName) throws JsonIOException, JsonSyntaxException,
+            JsonParseException, IOException, URISyntaxException, NoSuchFileException {
         JsonObject jsonObject = new JsonObject();
         JsonParser parser = new JsonParser();
-        JsonElement jsonElement = parser.parse(IOUtils.toString(fileName));
-        jsonObject = jsonElement.getAsJsonObject();
-        return jsonObject;
+        try (InputStream reader = fileName.openStream()) {
+            JsonElement jsonElement = parser.parse(new String(reader.readAllBytes(), StandardCharsets.UTF_8));
```

Comment:
While not a bad approach per se. In general you should avoid storing the entire file in memory if you would just pass it to another application for processing. Passing a reader for that file data would instead be preferable since it would allow the processing application to read and garbage collect data as it goes without requiring the entire file available at once.
```suggestion
        try (InputStream inputStream = fileName.openStream();
               InputStreamReader reader = new InputStreamReader(inputStream, StandardCharsets.UTF_8)) {
            JsonElement jsonElement = parser.parse(reader);
```